nixos/modules/virtualisation/glance.nix

```diff
@@ -0,0 +1,199 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.virtualisation.glance;
+  commonConf = ''
+    [database]
+    connection = mysql://glance:${cfg.dbPassword}@${cfg.dbHost}/glance
```

Comment:
@nlewo I don't think we have something like that. But there is known thing that all derivations end up in /nix/store so all the passwords go there.

What I do to avoid it is:
- I follow nixops pattern to put ALL credentials in the key store (https://nixos.org/nixops/manual/#opt-deployment.keys) - those keys are not going to be written in /nix/store, they will end up in /run/keys/<keyname>.
- in the scripts I use:
```
run-something --secret "$(</run/keys/<my-secret>)"
```
Config files are little harder. If it's possible to use that by the process, use mix of config files and environment variables. If it's not possible, I keep template of config file in nix store, and than, on preStart script I copy this file to /var/lib/<service>/ and fill the template (for example with sed) with keys from /run/keys.